tests/components/unifiprotect/test_event.py

```diff
@@ -152,3 +152,177 @@ def _capture_event(event: HAEvent) -> None:
     assert state
     assert state.state == timestamp
     unsub()
+
+
+async def test_doorbell_nfc_scanned(
+    hass: HomeAssistant,
+    ufp: MockUFPFixture,
+    doorbell: Camera,
+    unadopted_camera: Camera,
+    fixed_now: datetime,
+) -> None:
+    """Test a doorbell NFC scanned event."""
+
+    await init_entry(hass, ufp, [doorbell, unadopted_camera])
+    assert_entity_counts(hass, Platform.EVENT, 3, 3)
+    events: list[HAEvent] = []
+
+    @callback
+    def _capture_event(event: HAEvent) -> None:
+        events.append(event)
+
+    _, entity_id = ids_from_device_description(
+        Platform.EVENT, doorbell, EVENT_DESCRIPTIONS[1]
+    )
+
+    unsub = async_track_state_change_event(hass, entity_id, _capture_event)
+    event = Event(
+        model=ModelType.EVENT,
+        id="test_event_id",
+        type=EventType.NFC_CARD_SCANNED,
+        start=fixed_now - timedelta(seconds=1),
+        end=None,
+        score=100,
+        smart_detect_types=[],
+        smart_detect_event_ids=[],
+        camera_id=doorbell.id,
+        api=ufp.api,
+        metadata={"nfc": {"nfc_id": "test_nfc_id", "user_id": "test_user_id"}},
+    )
+
+    new_camera = doorbell.copy()
+    new_camera.last_nfc_card_scanned_event_id = "test_event_id"
+    ufp.api.bootstrap.cameras = {new_camera.id: new_camera}
+    ufp.api.bootstrap.events = {event.id: event}
+
+    mock_msg = Mock()
+    mock_msg.changed_data = {}
+    mock_msg.new_obj = event
+    ufp.ws_msg(mock_msg)
+
+    await hass.async_block_till_done()
+
+    assert len(events) == 1
+    state = events[0].data["new_state"]
+    assert state
+    assert state.attributes[ATTR_ATTRIBUTION] == DEFAULT_ATTRIBUTION
+    assert state.attributes[ATTR_EVENT_ID] == "test_event_id"
+    assert state.attributes.get("nfc_id") == "test_nfc_id"
```

Comment:
```suggestion
    assert state.attributes["nfc_id"] == "test_nfc_id"
```

Please use `[]` for expected values